### PR TITLE
Fix save and clear

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
add async await to formik that o submit miss async await

### CODE REVIEW

File: src/components/Shared/FormShell.js [hunk 1]
```diff
@@ -110,6 +112,26 @@ export default function FormShell({
     })
   }
 
+  const [saveAndClearSubmitted, setSaveAndClearSubmitted] = useState(false)
```
Comment (imtithalhalalo) [inline src/components/Shared/FormShell.js:115]: move the state to the top

File: src/components/Shared/FormShell.js [hunk 2]
```diff
@@ -110,6 +112,26 @@ export default function FormShell({
...
+
+  async function handleSaveAndClear() {
+    setSaveAndClearSubmitted(true)
+    form.submitForm()
+  }
+
+  useEffect(() => {}, [form])
```
Comment (imtithalhalalo) [inline src/components/Shared/FormShell.js:122]: what is it used for?

File: src/pages/batches/index.js [hunk 3]
```diff
@@ -25,341 +21,335 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
-    })
+  const documentTypesValidation = useFormik({
+    enableReinitialize: false,
+    validateOnChange: false,
 
-    const handleSubmit = () => {
-        if (activeTab === 0)
-            documentTypesValidation.handleSubmit()
+    validationSchema: yup.object({
+      reference: yup.string().required('This field is required'),
```
Comment (imtithalhalalo) [inline src/pages/batches/index.js:78]: remove the messages 'This field is required'

File: src/pages/batches/index.js [hunk 4]
```diff
@@ -25,341 +21,335 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
+    var parameters = defaultParams + '&_dgId=0'
+    getRequest({
+      extension: SystemRepository.DocumentType.qry,
+      parameters: parameters
+    })
+      .then(res => {
+        setGridData({ ...res, _startAt })
+      })
+      .catch(error => {
+        console.log({ error: error.response.data })
```
Comment (imtithalhalalo) [inline src/pages/batches/index.js:103]: do we need the console log here?

File: src/pages/batches/index.js [hunk 5]
```diff
@@ -25,341 +21,335 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
+    var parameters = '_database=25' //add 'xml'.json and get _database values from there
+    getRequest({
+      extension: SystemRepository.KeyValueStore,
+      parameters: parameters
+    })
+      .then(res => {
+        setSysFunctionsStore(res.list)
+      })
+      .catch(error => {
+        console.log({ error: error.response.data })
```
Comment (imtithalhalalo) [inline src/pages/batches/index.js:129]: also here for the log

File: src/pages/batches/index.js [hunk 6]
```diff
@@ -25,341 +21,335 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
+  const postDocumentType = async obj => {
+    const recordId = obj.recordId
+    await postRequest({
+      extension: SystemRepository.DocumentType.set,
+      record: JSON.stringify(obj)
+    })
+      .then(res => {
+        getGridData()
+        setWindowOpen(false)
+        if (!recordId) toast.success('Record Added Successfully')
```
Comment (imtithalhalalo) [inline src/pages/batches/index.js:168]: use platformLabels

File: src/pages/batches/index.js [hunk 7]
```diff
@@ -25,341 +21,335 @@ import { Grow } from 'src/components/Shared/Layouts/Grow'
...
-                console.log({ error: error })
-            })
-    }
+  const delDocumentType = obj => {
+    postRequest({
+      extension: SystemRepository.DocumentType.del,
+      record: JSON.stringify(obj)
+    })
+      .then(res => {
+        console.log({ res })
```
Comment (imtithalhalalo) [inline src/pages/batches/index.js:182]: remove console.log

**Comment by RebeccaFrem02** [comment]
image
when i click save and clear here the window closes and doesn't reopen

**Comment by RebeccaFrem02** [comment]
and also in pages that are not multitabs

**Comment by RebeccaFrem02** [comment]
if you click edit mode in a mutltitab form and you click save and clear the behaivior is correct but when you fill the form and you click the same button it behaives as you clicked submit

**Comment by Ghinwa-Arab** [comment]
image

Open new mode then click save & clear, fields validation doesn't trigger

**Comment by Ghinwa-Arab** [comment]
image

Add a new record and click save, then click save & clear without closing the window, the form not clearing.

**Comment by Ghinwa-Arab** [comment]
image

When you try to add a record with an existing name, you'll first receive an error, which is expected. However, after changing the name to a unique one and click 'save & clear', the form saves but doesn't clearing.

**Comment by Ghinwa-Arab** [comment]
If on save it returns any error the form shouldn't clear

**Comment by Ghinwa-Arab** [comment]
image

If save returns any error the form shouldn't clear

**Comment by imtithalhalalo** [comment]
Screenshot 2024-09-05 134629

It is not clearing after save

**Comment by imtithalhalalo** [comment]
Screenshot 2024-09-05 135431

'cb-cash-groups' it is not clearing after save

**Comment by Ghinwa-Arab** [comment]
image

Please check if something is affecting the save, as we're getting this error during 'save' or 'save & clear,' and we don't have reference here.

**Comment by Ghinwa-Arab** [comment]
image

'clear' and 'save & clear' buttons :
1. always should be in main tab only, please remove them from 2nd and 3rd tabs.
2. not clearing

**Comment by Ghinwa-Arab** [comment]
image

1. remove required for min unit price

**Comment by Ghinwa-Arab** [comment]
image

1. not clearing

**Comment by Ghinwa-Arab** [comment]
image

not clearing

**Comment by Alinadimmokdad** [comment]
in manufactoring , there is something wrong in : 
Routing save and clear just save 
works center also , save and groups
machines : open the previous window before editing + new mode issue 
labor page : save and clear not calling a function in edit mode 
Routing also open the previous window on edit + on add not clearing 
production Class also open the previous window on edit + on add not clearing 
Production Line : also open the previous window on edit + on add not clearing

**Pull request merged**